aws/resource_aws_elasticsearch_domain_test.go

```diff
@@ -77,6 +77,28 @@ func TestAccAWSElasticSearchDomain_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSElasticSearchDomain_ZoneAwarenessWith3AZ(t *testing.T) {
+	var domain elasticsearch.ElasticsearchDomainStatus
+	ri := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckESDomainDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccESDomainConfig_ZoneAwarenessWith3AZ(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckESDomainExists("aws_elasticsearch_domain.example", &domain),
+					resource.TestCheckResourceAttr(
+						"aws_elasticsearch_domain.example", "elasticsearch_version", "1.5"),
+					resource.TestMatchResourceAttr("aws_elasticsearch_domain.example", "kibana_endpoint", regexp.MustCompile(".*es.amazonaws.com/_plugin/kibana/")),
```

Comment:
Should this be testing the new attribute instead? e.g.

```suggestion
					resource.TestCheckResourceAttr("aws_elasticsearch_domain.example", "cluster_config.0.zone_awareness_count", "3"),
```

Reply:
Ok, i'm taking this recommendation and expanding it to multiple test steps:

1. zone awareness disabled and zone awareness count set to 3
2. update cluster with zone awareness enabled
3. update cluster again with zone awareness disabled
4. update cluster with zone awareness enabled and count set to 2
5. update cluster zone aware count to 3